src/App.tsx

```diff
@@ -15,48 +17,115 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortField {
+  sortFieldName = 'name',
+  sortFieldLength = 'length',
+  sortFieldNone = '',
+}
+
+type SortParams = {
+  sortField: SortField;
+  isReversed: boolean;
+};
+
+function sortGoodsByParams(
+  goods: string[],
+  { sortField, isReversed }: SortParams,
+): string[] {
+  const orderedField = [...goods];
+
+  if (sortField) {
+    orderedField.sort((good1, good2) => {
+      switch (sortField) {
+        case SortField.sortFieldName:
+          return good1.localeCompare(good2);
+
+        case SortField.sortFieldLength:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    orderedField.reverse();
+  }
+
+  return orderedField;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortField.sortFieldNone);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const visibleOrder = sortGoodsByParams(
+    goodsFromServer,
+    { sortField, isReversed },
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn(
+            'button is-info',
+            { 'is-light': sortField !== SortField.sortFieldName },
+          )}
+          onClick={() => setSortField(SortField.sortFieldName)}
```

Comment:
it is bad practise to make anonymous arrow functions in jsx.
instead make one fuction factory for both sort fields
```
const makeSetSortField = (field) => () => setSortField(field);
```
```suggestion
          onClick={makeSetSortField(SortField.sortFieldName)}
```